### PR TITLE
gh-398 prototype ui

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

File: ui/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+Option 2 - Build using the standalone-ui profile
+=============================================
+
+The application can be built and then run as a basic executable standalone war file from Maven. When run in this format, the example rest api is also deployed.
+
+To build it and its dependencies, use the following command from the parent directory:
+
+'mvn clean install -P standalone-ui'
```
Comment (gaffer01) [inline ui/README.md]: Is it worth adding `-DskipTests` to this line to speed it up?

File: ui/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+This uses the 'standalone-ui' profile to start a tomcat server with the ui and example-rest wars deployed.
+
+The ui can then be accessed via http://localhost:8080/ui and the rest api is at The ui can then be accessed via http://localhost:8080/rest
```
Comment (gaffer01) [inline ui/README.md]: Sentence starts twice: "and the rest api is at The ui can then be accessed".

Comment (p013570) [inline ui/README.md]: Good spot - thanks.

File: ui/src/main/webapp/app/controller.js [hunk 3]
```diff
@@ -0,0 +1,461 @@
+/*
```
Comment (gchqdev03) [inline ui/src/main/webapp/app/controller.js]: Would probably better to split this monolithic controller down a bit. Perhaps use angular services to encapsulate some of the logic and inject the service into the controller. 
Maybe even split the controller into multiple controllers that perform discrete functions

**Pull request merged**